test/honeybadger/http_adapter/req_test.exs

```diff
@@ -0,0 +1,114 @@
+defmodule Honeybadger.HTTPAdapter.ReqTest do
+  use Honeybadger.Case
+  doctest Honeybadger.HTTPAdapter.Req
+
+  alias Req.TransportError
+  alias Honeybadger.HTTPAdapter.{HTTPResponse, Req}
+
+  # Test retries quickly
+  @req_opts [retry_delay: 0]
+
+  describe "request/4" do
+    test "handles SSL" do
+      TestServer.start(scheme: :https)
+      TestServer.add("/", via: :get)
+
+      req_opts =
+        Keyword.put(
+          @req_opts,
+          :connect_options,
+          transport_opts: [cacerts: TestServer.x509_suite().cacerts],
+          protocols: [:http2]
+        )
+
+      assert {:ok, %HTTPResponse{status: 200, body: "HTTP/2"}} =
+               Req.request(:get, TestServer.url(), nil, [], req_opts)
+    end
```

Comment:
This is testing the functionality of `req`, and thereby `finch` and `mint`, which seems outside the scope of what the honeybadger library should care about.

Reply:
OK, I can see removing this test for that reason. I don't mind it being there to keep parity with other adapter test suites (e.g. hackney) but you're right that this one isn't testing much code in the library itself.